barman/recovery_executor.py

```diff
@@ -1807,13 +1810,327 @@ def _backup_copy(self, backup_info, dest, remote_command=None, **kwargs):
+            name -> location map (for relocation)
+        :param str|None remote_command: default None. The remote command to
+            recover the backup, in case of remote backup
+        """
+        # If it is a remote recovery we just follow the standard rsync copy process
+        if remote_command:
+            super(IncrementalRecoveryExecutor, self)._backup_copy(
+                backup_info, dest, tablespaces, remote_command, **kwargs
+            )
+            return
+        # If it is a local recovery we move the content from staging to destination
+        # Starts with tablespaces
+        if backup_info.tablespaces:
+            for tablespace in backup_info.tablespaces:
+                # By default a tablespace goes in the same location where
+                # it was on the source server when the backup was taken
+                destination = tablespace.location
+                # If a relocation has been requested for this tablespace
+                # use the user provided target directory
+                if tablespaces and tablespace.name in tablespaces:
+                    destination = tablespaces[tablespace.name]
+                # Move the content of the tablespace directory to destination directory
+                self._prepare_destination(destination)
+                tbs_source = backup_info.get_data_directory(
+                    tablespace_oid=tablespace.oid
+                )
+                self._move_to_destination(source=tbs_source, destination=destination)
+
+        # Then procede to move the content of the data directory
+        # We don't move the pg_tblspc as the _prepare_tablespaces method called earlier
+        # in the process already created this directory and required symlinks in the destination
+        data_source = backup_info.get_data_directory()
+        self._move_to_destination(
+            source=data_source, destination=dest, exclude_dirs=["pg_tblspc"]
+        )
+
+    def _move_to_destination(self, source, destination, exclude_dirs=[]):
+        """
+        Move a single directory or file to its destination directory
+
+        :param barman.infofile.SyntheticBackupInfo backup_info: the synthetic
+            backup info file
+        :param barman.infofile.Tablespace: the tablespace to be moved
+        :param dict[str,str]|None tablespaces_mapping: a tablespace
+            name -> location map (for relocation)
+        """
+        for file_or_dir in os.listdir(source):
+            if file_or_dir not in exclude_dirs:
+                file_or_dir_path = os.path.join(source, file_or_dir)
+                try:
+                    shutil.move(file_or_dir_path, destination)
+                except shutil.Error:
+                    output.error(
+                        "Destination directory '%s' must be empty." % destination
+                    )
+                    output.close_and_exit()
+
+    def _get_backup_chain_path(self, backup_info):
```

Comment:
```suggestion
    def _get_backup_chain_paths(self, backup_info):
```

It seems we are returning the path for each backup in the chain, so a plural sounds better to me here.